pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -156,26 +152,145 @@ def xyx_decomposition(U, wire, return_global_phase=False):
+        return_global_phase (bool): Whether to return the global phase
+            as a `qml.s_prod` between `exp(1j)*alpha` and `qml.Identity` as the last
+            element of the returned list of operations.
+
+    Returns:
+        list[Operation]: Returns a list of gates, an ``RZ``, an ``RX`` and
+        another ``RZ`` gate, which when applied in the order of appearance in the list is
+        equivalent to the unitary :math:`U` up to global phase. If `return_global_phase=True`,
+        the global phase is returned as the last element of the list.
+
+    **Example**
+
+    >>> U = np.array([[-0.28829348-0.78829734j,  0.30364367+0.45085995j],
+    ...               [ 0.53396245-0.10177564j,  0.76279558-0.35024096j]])
+    >>> decomp = zxz_decomposition(U, 0, return_global_phase=True)
+    >>> decomp
+        [RZ(tensor(-1.81289163, requires_grad=True), wires=[0]),
+        RX(tensor(1.14938178, requires_grad=True), wires=[0]),
+        RZ(tensor(-2.97933083, requires_grad=True), wires=[0]),
+        (0.38469215914523336-0.9230449299422961j)*(Identity(wires=[0]))]
+    """
+
+    # Small number to add to denominators to avoid division by zero
+    EPS = 1e-64
+
+    # Get global phase \alpha and U in SU(2) form (determinant is 1)
+    U = math.expand_dims(U, axis=0) if len(U.shape) == 2 else U
+    U_det1, alphas = _convert_to_su2(U, return_global_phase=True)
+
+    # Use top row to solve for \phi and \psi
+    phis_plus_psis = math.arctan2(-math.imag(U_det1[:, 0, 0]), math.real(U_det1[:, 0, 0]) + EPS)
+    phis_minus_psis = math.arctan2(-math.real(U_det1[:, 0, 1]), -math.imag(U_det1[:, 0, 1]) + EPS)
+    phis = phis_plus_psis + phis_minus_psis
+    psis = phis_plus_psis - phis_minus_psis
+
+    # Wrap angles to range [-\pi, \pi]
+    phis = (phis + qml.numpy.pi) % (2 * qml.numpy.pi) - qml.numpy.pi
+    psis = (psis + qml.numpy.pi) % (2 * qml.numpy.pi) - qml.numpy.pi
+
+    # Conditional to avoid divide by 0 errors
+    thetas = math.array(
+        [
+            2 * math.arccos(math.real(curr_U[0, 0]) / (math.cos(phi_plus_psi) + EPS))
+            if math.allclose(math.sin(phi_plus_psi), 0.0)
+            else 2 * math.arccos(-math.imag(curr_U[0, 0]) / (math.sin(phi_plus_psi) + EPS))
+            for curr_U, phi_plus_psi in zip(U_det1, phis_plus_psis)
+        ]
+    )
 
-    phi, theta, lam = map(math.squeeze, [phi, theta, lam])
+    phis, thetas, psis, alphas = map(math.squeeze, [phis, thetas, psis, alphas])
 
-    Operations = [qml.RX(phi, wire), qml.RY(theta, wire), qml.RX(lam, wire)]
+    # Return gates in the order they will be applied on the qubit
+    Operations = [qml.RZ(psis, wire), qml.RX(thetas, wire), qml.RZ(phis, wire)]
     if return_global_phase:
-        Operations += [qml.s_prod(math.exp(1j * gamma)[0], qml.Identity(wire))]
+        Operations += [qml.s_prod(math.exp(1j * alphas), qml.Identity(wire))]
 
     return Operations
+
+
+def one_qubit_decomposition(U, rotations, wire, return_global_phase=False):
```

Comment:
Could you add ZYZ as default?

Reply:
`wire` does not have a default argument, so if I added a default argument for `rotations`, I can't have a non-default argument following a default argument. What's the best way to deal with this?